src/app/active-mode/Views/current-activity/VendorBounties.tsx

```diff
@@ -1,110 +1,179 @@
-      vendorsResponse?.sales.data?.[vendorHash]?.saleItems,
-      {}
-    );
-    const vendorBounties = d2Vendor?.items.filter(
-      ({ item, canPurchase, canBeSold }: VendorItem) =>
-        canPurchase && canBeSold && item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties)
-    );
+  useEffect(() => {
+    if (!vendorsResponse) {
+      return;
+    }
+
+    const purchasableBounties: VendorItem[] = [];
+    const { vendors, sales, itemComponents } = vendorsResponse;
+
+    Object.values(vendors.data || []).forEach((vendor) => {
+      const { vendorHash } = vendor;
+      const d2Vendor = toVendor(
+        vendorHash,
+        defs,
+        buckets,
+        vendor,
+        account,
+        itemComponents[vendorHash],
+        sales.data?.[vendorHash]?.saleItems,
+        {}
+      );
+
+      d2Vendor &&
+        purchasableBounties.push(
+          ...d2Vendor.items.filter(
+            ({ item, canPurchase, canBeSold }: VendorItem) =>
+              canPurchase &&
+              canBeSold &&
+              item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties)
+          )
+        );
+    });
+
+    setBounties(purchasableBounties);
+  }, [defs, account, buckets, vendorsResponse]);
 
-    if (vendorBounties) {
-      bounties.push(...vendorBounties);
+  useEffect(() => {
+    setSuggestedBounties(bountiesForActivity(defs, bounties, activityInfo));
+  }, [defs, bounties, activityInfo]);
+
+  const ownedBountyHashes: number[] = [];
+  const unownedBounties: VendorItem[] = [];
+  suggestedBounties.forEach((vendorItem) => {
+    if (!vendorItem.item) {
+      return;
+    }
+    if (ownedItemHashes.has(vendorItem.item.hash)) {
+      ownedBountyHashes.push(vendorItem.item.hash);
+    } else {
+      unownedBounties.push(vendorItem);
     }
   });
 
+  const ownedIncompletePursuits = store.items
+    .filter(({ hash }) => ownedBountyHashes.includes(hash))
+    .filter(
+      ({ complete, pursuit }) =>
+        ((!complete && pursuit?.expirationDate?.getTime()) ?? 0) > Date.now()
+    );
+
   return (
     <>
-      {bounties?.map(
-        (item: VendorItem) =>
-          item.item &&
-          !ownedItemHashes.has(item.item.hash) && (
-            <VendorItemComponent
-              key={item.key}
-              defs={defs}
-              item={item}
-              owned={false}
-              characterId={store.id}
-            />
-          )
+      <div className={styles.bountyGuide}>
+        <PursuitsGroup
+          defs={defs}
+          store={store}
+          pursuits={ownedIncompletePursuits}
+          skipTypes={['ActivityMode', 'Destination']}
+          hideDescriptions
+        />
+      </div>
+      {unownedBounties.length > 0 && (
+        <>
+          <div className={styles.title}>{t('ActiveMode.SuggestedBounties')}</div>
+          <div className={styles.suggestedBounties}>
+            {unownedBounties.map(
+              (item: VendorItem) =>
+                item.item && (
```

Comment:
Unless we like showing the costs, it might be neat to just collect these as items and slap them in a PursuitsGroup as well